temporal/api/cloud/cloudservice/v1/request_response.proto

```diff
@@ -234,3 +234,103 @@ message GetRegionResponse {
     // The temporal cloud region.
     temporal.api.cloud.region.v1.Region region = 1;
 }
+
+message GetUserGroupsRequest {
+    // The requested size of the page to retrieve - optional.
+    // Cannot exceed 1000. Defaults to 100.
+    int32 page_size = 1;
+    // The page token if this is continuing from another response - optional.
+    string page_token = 2;
+    // Filter groups by the namespace they have access to - optional.
+    string namespace = 3;
+    // Filter groups by their type - optional.
+    // Allowed values ["google_workspace"].
+    string group_type = 4;
```

Comment:
Hmmmm why do we need this? Isn't this sort of invisible to them?

Reply:
we want this for extensibility. say a company was on google_workspace but wants to switch to some other IdP. we might need to support 2 sets of groups with the same name. group_type is there to differentiate them

Reply:
I was thinking do we need an abstract primary key for groups and then also maintain that group name + group type are unique?

My preference is to not name this "group type" since it's so generic. Instead something more descriptive like "idp_type" would. Though the other question I have is--is this supposed to eventually be used for temporal only groups?

Reply:
discussed offline and will remove group type for now. we can always add back in later